### PR TITLE
http client implementation using cross platform yhirose/cpp-httplib library

### PR DESCRIPTION
**Pull request by lalitb**
This PR provides header only http client implementation for HTTP client API (refer  PR #370 ), internally using cross-platform header-only [httplib](https://github.com/yhirose/cpp-httplib) library.

TBD - Unit test case, thread safe.

Please review it, while I would be adding TBD changes as part of this. This PR is dependent on #370

### CODE REVIEW

File: exporters/include/opentelemtry/exporters/common/http/httplib.h [hunk 1]
```diff
@@ -0,0 +1,8140 @@
+//
+//  httplib.h
+//
+//  Copyright (c) 2020 Yuji Hirose. All rights reserved.
+//  MIT License
+//
+
+#ifndef CPPHTTPLIB_HTTPLIB_H
```
Comment (jsuereth) [inline exporters/include/opentelemtry/exporters/common/http/httplib.h:8]: How is this library distributed?   

should we "shade" this library behind a namespace so we don't wind up with version issues if someone pulls in a  different version that is somehow not source compatible?

**Comment by maxgolov** [comment]
I would suggest to abandon this and work either on:
- in-house HTTP client (for Unit tests); or, better
- `libevent` -based HTTP client implementation

We probably don't want to add this MIT-licensed code to our repo, as well as take dependency on it - it does not seem to be widely used in production.

**Comment by lalitb** [comment]
We are no longer going to use httplib as client library.

**Comment by reyang** [comment]
Superseded by #385.

**Pull request closed without merging**